### ISSUE TITLE
Constant mic usage (orange dot in Monterey)

### ISSUE THREAD

**Issue by wpbdry** [kyleneideck/BackgroundMusic#562]
Background Music seems to be constantly using my mic? Why is this necessary?

Apart from any privacy concerns this is annoying as MacOS Monterey now displays an orange dot whenever an application is using the mic. This is visible constantly, even when watching a video in full-screen.

I am aware that Background Music creates a virtual audio input device which can be used to record computer audio. But I was not aware that this amounts to "using the mic" (my headphones do not create the dot when they are connected). Background Music should not need to use the built in mic or any other audio input device for any reason, as this software is primarily concerned with monitoring and tweaking audio output..

Please fix or help me understand! Thanks!

<img width="310" alt="image" src="https://user-images.githubusercontent.com/41445151/144789628-fda92585-2764-40d8-9f19-9cca452b9f2e.png">

**Comment by kyleneideck** [comment]
Background Music doesn't use any of your real input devices. macOS is showing the orange dot because Background Music uses the "Background Music" virtual input device.

I guess maybe their logic is that a virtual input device might still produce audio that should be private. @q-p reported it to Apple as a bug, but they said it was intentional: https://github.com/q-p/SoundPusher/issues/20

The best way to fix it would be for BGMDriver to send the audio to BGMApp through shared memory instead of a virtual input device. That would probably use less CPU and reduce the latency a bit, too.

But it's not a small amount of work and it's not likely I'll have time to work on it any time soon. If anyone else is interested in working on it, I'd be happy to help.

**Comment by q-p** [comment]
Here's a work-around for hiding the dot (but personally speaking I would not include such a hack in [SoundPusher](https://github.com/q-p/SoundPusher) as it might also hide "real" microphone dots) https://github.com/s4y/undot

**Comment by anglee2002** [comment]
feel sorry to see that, beacause of this issue, I diecide choose another application though

**Comment by lowking** [comment]
> feel sorry to see that, beacause of this issue, I diecide choose another application though

Which one? Give me a link, thanks

**Comment by anglee2002** [comment]
> > feel sorry to see that, beacause of this issue, I diecide choose another application though
> 
> Which one? Give me a link, thanks

sorry bro,haven’t found a good one perfectly fit my need.

**Comment by bauerp** [comment]
Yeah seeing orange dots or icons is a no go for me too, and I'm not about to install something to get rid of it.
I use Kap. Records system audio as well and no dots or icons unless I record.
https://github.com/wulkano/Kap

**Comment by terax6669** [comment]
I don't understand the issue. So the icon is there simply because the app has requested access to the microphone? Sounds like that shouldn't be the case.

Notification should appear only when the microphone is **actually** being accessed. I don't see this icon simply because I have Teams open... it appears only when I'm on a call.

**Comment by kyleneideck** [comment]
@terax6669 Background Music sends the system audio to its app through its virtual input device. That used to work well enough, but now the orange dot appears when the app accesses the virtual input device. That happens because macOS doesn't care whether it's a virtual input device or a real one, e.g. a microphone. Background Music doesn't access your microphone or any other input devices except its own.

I've been working on a fix for this issue by replacing the virtual input device with shared memory, but I'm not sure when it'll be ready. It's quite a large change.

If you're seeing the orange dot even when you're not playing audio, that's probably a bug in Background Music, possibly #470. I actually think I might have figured that issue out while working on the fix for this one, but that's still a work in progress too.

**Comment by gchilds** [comment]
If you're using BGM to adjust application audio volumes then from macOS's point of view BGM is recording audio.

**Comment by gchilds** [comment]
There's a newer API to for doing application audio filtering (see `CATap`) but you'll still get an orange dot of shame.

**Comment by terax6669** [comment]
> If you're using BGM to adjust application audio volumes then from macOS's point of view BGM is recording audio.

Thank you for the clarification

**Comment by mayurankv** [comment]
Just wondering if there is any update on this - currently not ideal to use with the icon always showing

EDIT: P.s. thank you for all your work and the amazing project :))
